src/schema/validator.ts

```diff
@@ -0,0 +1,185 @@
+    }
+    if (conditions.client_filters !== undefined) {
+        validateClientFilters(conditions.client_filters);
+    }
+}
+
+function validateClientFilters(client_filters: any) {
+    if (!Array.isArray(client_filters)) {
+        throw new TypeError("Client filter Collection must be an array.");
+    }
+
+    for (const filter of client_filters) {
+        if (typeof filter.name !== "string") {
+            throw new TypeError("Client filter name must be a string.");
+        }
+        if (filter.parameters !== undefined && typeof filter.parameters !== "object") {
+            throw new TypeError("Client filter parameters must be an object.");
+        }
+        // validate parameters here if we have schema of specific filters in the future
+    }
+}
+
+function validateVariants(variants: any) {
+    if (!Array.isArray(variants)) {
+        throw new TypeError("Variants must be an array.");
+    }
+
+    for (const variant of variants) {
+        if (typeof variant.name !== "string") {
+            throw new TypeError("Variant name must be a string.");
+        }
+        // skip configuration_value validation as it accepts any type
+        if (variant.status_override !== undefined && typeof variant.status_override !== "string") {
+            throw new TypeError("Variant status override must be a string.");
+        }
+        if (variant.status_override !== undefined && variant.status_override !== "None" && variant.status_override !== "Enabled" && variant.status_override !== "Disabled") {
+            throw new TypeError("Variant status override must be 'None', 'Enabled', or 'Disabled'.");
+        }
+    }
+}
+
+// #region Allocation
+function validateVariantAllocation(allocation: any) {
+    if (typeof allocation !== "object") {
+        throw new TypeError("Variant allocation must be an object.");
+    }
+
+    if (allocation.default_when_disabled !== undefined && typeof allocation.default_when_disabled !== "string") {
+        throw new TypeError("Variant allocation default_when_disabled must be a string.");
+    }
+    if (allocation.default_when_enabled !== undefined && typeof allocation.default_when_enabled !== "string") {
+        throw new TypeError("Variant allocation default_when_enabled must be a string.");
+    }
+    if (allocation.user !== undefined) {
+        validateUserVariantAllocation(allocation.user);
+    }
+    if (allocation.group !== undefined) {
+        validateGroupVariantAllocation(allocation.group);
+    }
+    if (allocation.percentile !== undefined) {
+        validatePercentileVariantAllocation(allocation.percentile);
+    }
+    if (allocation.seed !== undefined && typeof allocation.seed !== "string") {
+        throw new TypeError("Variant allocation seed must be a string.");
+    }
+}
+
+function validateUserVariantAllocation(UserAllocations: any) {
+    if (!Array.isArray(UserAllocations)) {
+        throw new TypeError("User allocations must be an array.");

```

Comment:
How about "'user allocation must be an array" or "Allocation "user" collection must be an array